### PR TITLE
Mining switch

### PR DESCRIPTION
**Pull request by h4x3rotab**

### CODE REVIEW

File: pallets/phala/src/mock.rs [hunk 1]
```diff
@@ -130,14 +131,17 @@ impl mining::Config for Test {
 	type OnReclaim = PhalaStakePool;
 	type OnStopped = PhalaStakePool;
 	type OnTreasurySettled = ();
+	type UpdateTokenomicOrigin = frame_system::EnsureRoot<u64>;
```
Comment (jasl) [inline pallets/phala/src/mock.rs:134]: `frame_system::EnsureRoot<Self::AccountId>` is better

File: pallets/phala/src/mock.rs [hunk 2]
```diff
@@ -130,14 +131,17 @@ impl mining::Config for Test {
...
 }
 
 impl stakepool::Config for Test {
 	type Event = Event;
 	type Currency = Balances;
 	type MinContribution = MinContribution;
 	type GracePeriod = MiningGracePeriod;
+	type MiningEnabledDefault = MiningEnabledDefault;
 	type OnSlashed = ();
+	type MiningSwitchOrigin = frame_system::EnsureRoot<u64>;
```
Comment (jasl) [inline pallets/phala/src/mock.rs:144]: `frame_system::EnsureRoot<Self::AccountId>` is better

File: pallets/phala/src/mock.rs (hunk 1 above)
Comment (h4x3rotab) [reply pallets/phala/src/mock.rs]: Done

File: pallets/phala/src/stakepool.rs [hunk 3]
```diff
@@ -47,17 +47,24 @@ pub mod pallet {
...
 		#[pallet::constant]
 		type MinContribution: Get<BalanceOf<Self>>;
 
 		/// The grace period for force withdraw request, in seconds.
 		#[pallet::constant]
 		type GracePeriod: Get<u64>;
 
+		/// If mining is enabled by default.
+		#[pallet::constant]
+		type MiningEnabledDefault: Get<bool>;
```
Comment (jasl) [inline pallets/phala/src/stakepool.rs:61]: `MiningEnabledByDefault` is better?

File: pallets/phala/src/mock.rs (hunk 2 above)
Comment (h4x3rotab) [reply pallets/phala/src/mock.rs]: Done

File: pallets/phala/src/stakepool.rs (hunk 3 above)
Comment (h4x3rotab) [reply pallets/phala/src/stakepool.rs]: Done

**Pull request merged**